pkg/estimator/client/interface.go

```diff
@@ -30,26 +30,40 @@ import (
 const UnauthenticReplica = -1
 
 var (
-	replicaEstimators              = map[string]ReplicaEstimator{}
-	unschedulableReplicaEstimators = map[string]UnschedulableReplicaEstimator{}
+	replicaEstimators              []ReplicaEstimator
+	unschedulableReplicaEstimators []UnschedulableReplicaEstimator
 )
 
 // ReplicaEstimator is an estimator which estimates the maximum replicas that can be applied to the target cluster.
 type ReplicaEstimator interface {
 	MaxAvailableReplicas(ctx context.Context, clusters []*clusterv1alpha1.Cluster, replicaRequirements *workv1alpha2.ReplicaRequirements) ([]workv1alpha2.TargetCluster, error)
+	GetPriority() EstimatorPriority
 }
 
 // UnschedulableReplicaEstimator is an estimator which estimates the unschedulable replicas which belong to a specified workload.
 type UnschedulableReplicaEstimator interface {
 	GetUnschedulableReplicas(ctx context.Context, clusters []string, reference *workv1alpha2.ObjectReference, unschedulableThreshold time.Duration) ([]workv1alpha2.TargetCluster, error)
+	GetPriority() EstimatorPriority
 }
 
 // GetReplicaEstimators returns all replica estimators.
-func GetReplicaEstimators() map[string]ReplicaEstimator {
+func GetReplicaEstimators() []ReplicaEstimator {
 	return replicaEstimators
 }
 
 // GetUnschedulableReplicaEstimators returns all unschedulable replica estimators.
-func GetUnschedulableReplicaEstimators() map[string]UnschedulableReplicaEstimator {
+func GetUnschedulableReplicaEstimators() []UnschedulableReplicaEstimator {
 	return unschedulableReplicaEstimators
 }
+
+// EstimatorPriority
+// 1. If higher-priority estimator has given the result, no longer to call lower-priority estimator.
+// 2. If two estimators are of the same priority, call both and choose the minimum value of each estimated result.
+// 3. If an estimator can only give a part of result of member clusters, same priority estimator will be called, but low-priority estimator not.
+type EstimatorPriority int32
+
+const (
+	Rough    EstimatorPriority = 100
+	General  EstimatorPriority = 200
+	Accurate EstimatorPriority = 300
```

Comment:
Do these numbers mean anything? If not, you can define a default priority in the range 0 to 100.

Reply:
No special meaning, but intentionally enumerated from the hundreds, so that if you want to add a priority between the General and Accurate levels, just give him a value between 200 and 300.